nucleus/dataset.py

```diff
@@ -1862,3 +1851,75 @@ def query_items(self, query: str) -> Iterable[DatasetItem]:
         )
         for item_json in json_generator:
             yield DatasetItem.from_json(item_json)
+
+    def create_track(
+        self,
+        reference_id: str,
+        scene_reference_id: Optional[str] = None,
+        metadata: Optional[dict] = None,
+    ) -> Track:
+        """
+        Creates a track with a reference_id and metadata, optionally attaching it to a scene.
+
+        Parameters:
+            reference_id (str): The user-defined reference to the track.
+                Must be unique per dataset.
+            scene_reference_id (Optional[str]): The user-defined reference to the scene this track
+                pertains to.
+            metadata (Optional[dict]): An arbitrary dictionary of additional data about this track that can be stored
+                and retrieved.
+
+        Returns:
+            The created `Track` object.
+        """
```

Comment:
Are we expecting tracks on dataset items that aren't scenes? 

I think we should go for an immutable model for tracks. On creating a track should have references to all the objects belonging to it.